File: sdk/src/main/java/com/sygic/travel/sdk/model/api/MediaResponse.java
package com.sygic.travel.sdk.model.api;

import com.google.gson.JsonObject;
import com.google.gson.annotations.SerializedName;
import com.google.gson.reflect.TypeToken;
import com.sygic.travel.sdk.Parser;
import com.sygic.travel.sdk.model.media.Medium;

import java.util.List;

/**
 * <p>Response containing a list of place media. Suitable for a gallery.</p>
 */
public class MediaResponse extends StResponse {
	private static final String MEDIA = "media";
	private List<Medium> media;

	private JsonObject data;

	public Object getData() {
		if(media == null && data != null) {
			media = Parser.parseJson(
				data.get(MEDIA).toString(),
				new TypeToken<List<Medium>>(){}.getType()
			);
		}
		return media;
	}
}

File: sdk/src/main/java/com/sygic/travel/sdk/places/responseWrappers/MediaResponse.kt
package com.sygic.travel.sdk.places.responseWrappers

import com.google.gson.annotations.SerializedName
import com.sygic.travel.sdk.common.responseWrappers.StResponse
import com.sygic.travel.sdk.places.model.api.media.ApiMedium
import com.sygic.travel.sdk.places.model.media.Medium

/**
 * Response containing a list of place media. Suitable for a gallery.
 */
class MediaResponse : StResponse() {
	private var data: Data? = null

	fun getMedia(): List<Medium>? {
		val media = data?.apiMedia
		val convertedMedia: MutableList<Medium> = mutableListOf()

		media?.mapTo(convertedMedia) {
			it.convert()
		}

		return convertedMedia
	}

	inner class Data {
		@SerializedName("media")
		internal var apiMedia: List<ApiMedium>? = null
	}
}
